src/dotnet/CommandLine/CommandLineApplication.cs

```diff
@@ -129,10 +130,18 @@ public int Execute(params string[] args)
                     if (longOption != null)
                     {
                         processed = true;
-                        option = command.Options.SingleOrDefault(opt => string.Equals(opt.LongName, longOption[0], StringComparison.Ordinal));
+                        string longOptionName = longOption[0];
+                        option = command.Options.SingleOrDefault(opt => string.Equals(opt.LongName, longOptionName, StringComparison.Ordinal));
 
                         if (option == null)
                         {
+                            if (string.IsNullOrEmpty(longOptionName) && !command._throwOnUnexpectedArg && AllowArgumentSeparator)
```

Comment:
Correct, but I felt it more correct to use `string.IsNullOrEmpty` instead of longOptionName.Length == 0, in case that assumption ever changes.